src/CTA.Rules.Models/Actions/ExpressionAction.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace CTA.Rules.Models
+{
+    public class ExpressionAction : GenericAction
+    {
+        public Func<SyntaxGenerator, SyntaxNode, SyntaxNode> ExpressionActionFunc { get; set; }
+        public override bool Equals(object obj)
+        {
+            var action = (ExpressionAction)obj;
+            return action.Key == this.Key
+                && action.Value == this.Value
+                && action.ExpressionActionFunc.Method.Name == this.ExpressionActionFunc.Method.Name;
```

Comment:
Any null checks needed here?

Reply:
Also added null checks to be on the safe side and across the other objects too.